src/pages/overlay/hooks/useSleeping.tsx

```diff
@@ -63,12 +65,28 @@ export const SleepingProvider = ({ children }: { children: ReactNode }) => {
     stopTimer();
   }, [callbacks.sleep, stopTimer]);
 
+  // Pause the timer and keep the overlay awake
+  const stayAwake = useCallback(() => {
+    setSleeping(false);
+    stopTimer();
+  }, [stopTimer]);
```

Comment:
Should this block calls to wake/sleep, which could cause it to sleep, while stay awake is active?